addons/mail/static/src/core/common/message.js

```diff
@@ -285,7 +285,12 @@ export class Message extends Component {
      * @returns {boolean}
      */
     get canToggleStar() {
-        return Boolean(!this.message.is_transient && this.message.res_id && this.store.user);
+        return Boolean(
+            !this.message.is_transient &&
+                this.message.res_id &&
+                this.store.user &&
+                !this.props.thread?.isTransient
+        );
```

Comment:
Would change `!this.props.thread?.isTransient` to `Number.isInteger(!this.props.message.id)`, so that we have better guarantee this targets a genuine message on server which is what's required to start and mark as unread.

`props.isTransient` requires more LOCs to maintain this logic in client-side code, and also works as long as we have thread data (in theory no guarantees we have both, even though I don't think this can happen in practice currently with transient messages).